### PR TITLE
Extract `AppBuilder` to separate module

### PR DESCRIPTION
**Pull request by DariuszDepta**
To reduce the `App.rs` file size, extracted the `AppBuilder` to separate module. It simplifies analysing both of them.

### CODE REVIEW

File: tests/app_builder/with_bank.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        todo!()
+    }
+}
+
+impl Bank for MyBankKeeper {}
+
+#[test]
+fn building_app_with_custom_bank_should_work() {
+    let app_builder = AppBuilder::default();
+    let _ = app_builder.with_bank(MyBankKeeper {}).build(|_, _, _| {});
```
Comment (chipshort) [inline tests/app_builder/with_bank.rs:65]: Shouldn't these tests assert something? E.g. in this case sending some Bank message and verifying that it is actually sent to `MyBankKeeper`?

File: tests/app_builder/with_bank.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    Addr, Api, BankMsg, BankQuery, Binary, BlockInfo, CustomQuery, Querier, Storage,
+};
+use cw_multi_test::{AppBuilder, AppResponse, Bank, BankSudo, CosmosRouter, Module};
+use schemars::JsonSchema;
+use serde::de::DeserializeOwned;
+use std::fmt::Debug;
+
+struct MyBankKeeper {}
+
+impl Module for MyBankKeeper {
```
Comment (chipshort) [inline tests/app_builder/with_bank.rs:12]: All of the tests share the same implementation, so I would just use the same keeper in all of them.
That will shorten the code significantly (and maybe it makes more sense at that point to move the tests to the same file as the keeper)

Comment (DariuszDepta) [reply tests/app_builder/with_bank.rs]: I have removed all bloated code that in fact would duplicate tests we already have in every single module or keeper. I just simplified the test cases, just to make sure that the `App` is build and initialized using `AppBuilder` (that was the goal of introducing these tests).

File: tests/app_builder/with_bank.rs (hunk 1 above)
Comment (chipshort) [reply tests/app_builder/with_bank.rs]: But this test only ensures that the `App` is built, not that it actually uses the given keeper.

File: tests/app_builder/with_block.rs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    let mut initialized = false;
+    let app_builder = AppBuilder::default();
+    let _ = app_builder
+        .with_block(BlockInfo {
+            height: 20,
+            time: Timestamp::from_nanos(1_571_797_419_879_305_544),
+            chain_id: "my-testnet".to_string(),
+        })
+        .build(|_, _, _| {
+            initialized = true;
```
Comment (chipshort) [inline tests/app_builder/with_block.rs:15]: Keeper tests are looking great now! I think this one should also assert the block info.

File: tests/app_builder/with_storage.rs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+use cw_multi_test::AppBuilder;
+
+#[test]
+fn building_app_with_custom_storage_should_work() {
+    let mut initialized = false;
+    let app_builder = AppBuilder::default();
+    let _ = app_builder
+        .with_storage(MockStorage::default())
+        .build(|_, _, _| {
+            initialized = true;
```
Comment (chipshort) [inline tests/app_builder/with_storage.rs:11]: IMHO, this one should also use a custom storage and assert that it was actually called.

File: tests/app_builder/with_bank.rs (hunk 1 above)
Comment (DariuszDepta) [reply tests/app_builder/with_bank.rs]: Added tests that ensure, that custom keeper is called.

**Pull request merged**